noir-projects/noir-contracts/contracts/inclusion_proofs_contract/src/test.nr

```diff
@@ -0,0 +1,366 @@
+
+    // Each of these tests (note inclusion, note non-nullification, and validity (inclusion & non-nullification)) check the assertion at the block of creation of note, as well as at the "current" block
+    InclusionProofs::at(contract_address)
+        .test_note_inclusion(owner, true, note_creation_block_number, false)
+        .call(&mut env.private());
+    InclusionProofs::at(contract_address).test_note_inclusion(owner, false, 0, false).call(
+        &mut env.private(),
+    );
+
+    InclusionProofs::at(contract_address)
+        .test_note_not_nullified(owner, true, note_creation_block_number, false)
+        .call(&mut env.private());
+    InclusionProofs::at(contract_address).test_note_not_nullified(owner, false, 0, false).call(
+        &mut env.private(),
+    );
+
+    InclusionProofs::at(contract_address)
+        .test_note_validity(owner, true, note_creation_block_number, false)
+        .call(&mut env.private());
+    InclusionProofs::at(contract_address).test_note_validity(owner, false, 0, false).call(
+        &mut env.private(),
+    );
+}
+
+#[test]
+unconstrained fn nullify_note_flow() {
+    // This test creates a note, nullifies it, and checks certain properties about it, namely its inclusion in the trees,
+    // the non-inclusion of its nullifier, and its validity (the combination of the previous two). These properties are checked before nullification.
+    let (env, contract_address, owner) = setup(INITIAL_VALUE);
+    env.impersonate(owner);
+
+    InclusionProofs::at(contract_address).create_note(owner, 5).call(&mut env.private());
+
+    env.advance_block_by(1);
+
+    let note_valid_block_number = env.block_number();
+
+    InclusionProofs::at(contract_address).nullify_note(owner).call(&mut env.private());
+
+    env.advance_block_by(1);
+
+    // We test note inclusion at the note creation block and at current block
+    InclusionProofs::at(contract_address)
+        .test_note_inclusion(owner, true, note_valid_block_number, true)
+        .call(&mut env.private());
+
+    InclusionProofs::at(contract_address).test_note_inclusion(owner, false, 0, true).call(
+        &mut env.private(),
+    );
+
+    // We test note non-nullification and validity at the note creation block
+    InclusionProofs::at(contract_address)
+        .test_note_not_nullified(owner, true, note_valid_block_number, true)
+        .call(&mut env.private());
+    InclusionProofs::at(contract_address)
+        .test_note_validity(owner, true, note_valid_block_number, true)
+        .call(&mut env.private());
+}
+
+#[test(should_fail_with = "Assertion failed: Proving nullifier non-inclusion failed: low_nullifier.value < nullifier.value check failed")]
+unconstrained fn note_not_nullified_after_nullified() {
+    let (env, contract_address, owner) = setup(INITIAL_VALUE);
+    env.impersonate(owner);
+
+    InclusionProofs::at(contract_address).create_note(owner, 5).call(&mut env.private());
+
+    env.advance_block_by(1);
```

Comment:
Shouldn't we automine now? And if it's not yet implemente do we have an issue?

If we have an issue would include a todo everywhere here.